scripts/tools/zap/generate.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import json
+import os
+from pathlib import Path
+import sys
+
+# Check python version
+if sys.version_info[0] < 3:
+    print('Must use Python 3. Current version is ' + str(sys.version_info[0]))
+    exit(1)
+
+argc = len(sys.argv) - 1
+if argc == 0:
+  print('This script must have at least one argument, got', argc)
+  exit(1)
+
+if argc > 2:
+  print('This script must have at most two arguments, got', argc)
+  exit(1)
+
+CHIP_ROOT_DIR = os.path.realpath(os.path.join(os.path.dirname(__file__), '../../..'))
+os.chdir(CHIP_ROOT_DIR)
+if not os.path.exists(os.path.join(CHIP_ROOT_DIR, "scripts/tools/zap/generate.py")):
+    print('This script must be called from the root of the chip directory')
+    exit(1)
+
+APP_ZAP_FILE = os.path.join(CHIP_ROOT_DIR, sys.argv[1])
+if not os.path.exists(APP_ZAP_FILE) or not os.path.isfile(APP_ZAP_FILE):
+    print('Specified .zap file does not exists or is not a file')
+    exit(1)
+
+CHIP_TEMPLATES_DIR = os.path.join(CHIP_ROOT_DIR, "src", "app", "zap-templates")
+APP_ROOT_DIR = Path(APP_ZAP_FILE).parent
+
+# By default, this script assumes that the global CHIP template is used with
+# a default 'gen/' output folder relative to APP_ROOT_DIR.
+# If needed, the user may specify a specific template as a second argument. In
+# this case the output folder is relative to CHIP_ROOT_DIR.
+APP_ZAP_TEMPLATE = os.path.join(CHIP_TEMPLATES_DIR, "app-templates.json")
+APP_OUTPUT_DIR = os.path.join(APP_ROOT_DIR, "gen")
+
+if argc == 2:
+  APP_ZAP_TEMPLATE = os.path.join(CHIP_ROOT_DIR, sys.argv[2])
+  APP_OUTPUT_DIR = CHIP_ROOT_DIR
+
+if not os.path.exists(APP_ZAP_TEMPLATE) or not os.path.isfile(APP_ZAP_TEMPLATE):
+    print(APP_ZAP_TEMPLATE, 'does not exists or is not a file')
+    exit(1)
+
+if not os.path.exists(APP_OUTPUT_DIR) or not os.path.isdir(APP_OUTPUT_DIR):
+    print(APP_OUTPUT_DIR, 'does not exists or is not a directory')
+    exit(1)
+
+# Generate the files
+CHIP_ZAPRUNNER_DIR = os.path.join(CHIP_ROOT_DIR, "third_party", "zap", "repo")
+CHIP_ZCL = os.path.join(CHIP_TEMPLATES_DIR, "zcl", "zcl.json")
+
+os.chdir(CHIP_ZAPRUNNER_DIR)
+os.system('node ./src-script/zap-generate.js -z ' + CHIP_ZCL + ' -g ' + APP_ZAP_TEMPLATE + ' -i ' + APP_ZAP_FILE + ' -o ' + APP_OUTPUT_DIR)
```

Comment:
consider adding quotes.
Can also use subprocess and use list arrays and not worry about quoting.